build_runner/lib/src/generate/build_impl.dart

```diff
@@ -205,8 +205,7 @@ class BuildImpl {
   Set<AssetId> _matchingInputs(InputSet inputSet, int phaseNumber) =>
       _inputsForPhase(phaseNumber)
           .where((input) =>
-              input.package == inputSet.package &&
-              inputSet.globs.any((g) => g.matches(input.path)))
+              input.package == inputSet.package && inputSet.matches(input.path))
```

Comment:
How about `inputSet.match(input)` and shift the package checking there as well?

Reply:
Love it. PTAL @natebosch